model_compression_toolkit/core/common/model_collector.py

```diff
@@ -154,14 +184,35 @@ def infer(self, inputs_list: List[np.ndarray]):
 
         # TODO: Thinking about delegating collections to framework
         # TODO: migrate datasets to framework datasets
-        tensor_data = self.fw_impl.run_model_inference(self.model, inputs_list)
-        for td, sc in zip(tensor_data, self.stats_containers_list):
+        compute_hessians = self.hessian_service is not None and self.qc.activation_error_method == QuantizationErrorMethod.HMSE
```

Comment:
Should an error be thrown if the user passed HMSE but the hessian service is None?